In [3]:
import numpy as np 
import pandas as pd 

In [4]:
from sklearn.model_selection import train_test_split

# Load iris data
water_data = pd.read_csv('water_potability.csv')

# Extract features (X) and target labels (y)
y = water_data['Potability'].values
X = water_data.drop('Potability', axis=1).values

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Print the shapes of the training and testing sets to verify the split
print("Shape of X_train:", X_train.shape)
print("Shape of X_test:", X_test.shape)
print("Shape of y_train:", y_train.shape)
print("Shape of y_test:", y_test.shape)

Shape of X_train: (2620, 9)
Shape of X_test: (656, 9)
Shape of y_train: (2620,)
Shape of y_test: (656,)


In [5]:
print(y_test)

[0 1 0 0 1 1 0 0 0 0 0 1 0 1 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0
 0 0 1 0 1 1 0 0 1 1 0 0 0 1 1 0 1 0 0 0 0 0 0 0 1 1 0 1 0 1 0 1 1 0 1 0 0
 0 1 1 1 0 0 1 1 0 0 1 0 0 1 0 0 1 0 1 0 0 0 0 0 0 1 1 0 1 0 0 0 0 0 1 0 0
 0 0 0 0 1 1 0 1 1 0 0 0 0 0 1 0 1 0 0 1 1 0 0 1 1 0 1 0 1 0 0 1 0 0 0 1 1
 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 1 0 0 1 1 1 0 1 0 1 1 1 1 0 0
 0 1 0 1 1 1 0 0 1 1 0 0 1 0 0 1 1 0 0 0 0 0 1 1 1 0 0 0 1 0 0 1 1 0 0 0 1
 0 0 0 0 0 1 0 1 0 0 0 1 0 1 0 0 0 0 0 0 0 0 1 0 1 1 0 0 0 1 0 0 1 0 0 0 0
 1 0 1 0 1 1 0 1 1 0 1 1 0 1 0 0 0 0 1 0 0 0 0 1 0 0 1 1 1 0 0 0 1 1 1 0 0
 1 1 1 1 0 1 0 1 0 0 0 1 1 0 1 0 0 0 0 1 1 1 0 0 0 0 0 1 1 1 0 1 0 1 0 0 0
 0 0 0 0 1 0 0 0 0 1 0 1 1 1 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1 1 0 0 0 0 1 0 1
 0 1 0 0 0 1 0 0 1 1 0 1 1 1 1 0 0 0 0 1 0 0 1 0 1 0 0 0 1 0 0 0 0 0 0 0 1
 1 1 1 0 1 0 0 0 1 0 1 0 1 0 1 1 0 1 0 0 1 1 0 1 0 1 1 0 1 0 0 0 1 0 0 1 0
 1 0 1 0 0 1 0 0 0 0 0 0 0 1 1 1 1 1 0 0 0 0 1 1 0 0 1 0 0 1 1 0 0 0 0 0 0
 1 0 0 0 0 0 0 0 1 1 1 1 

## Training a simple Decision Tree model to predict water potability

In [6]:
from dt_exer import DecisionTreeClassifier

# Instantiate the classifier
clf = DecisionTreeClassifier(max_depth = 3)
print("Made classifier")

# Train the classifier
clf.fit(X_train, y_train)

print("fitting....")

# Make predictions on the test data
y_pred = clf.predict(X_test)
print("predicting....")


# Compare the predicted values to the actual labels and calculate f1 score
from sklearn.metrics import f1_score
f1 = f1_score(y_test, y_pred, average='weighted')
print("F1 Score:", f1)

Made classifier
fitting....
predicting....
F1 Score: 0.5205374913880919


### Results of this basic first test:
- A single decision tree has a 0.52 f1 score
- Takes around 1 minute to train the decision tree
  - This is probably due to the fact that there are multiple features to search through and the class
    that I made from scratch is not optimized

## Training a Random Forest model to predict water potability